src/network/mod.rs

```diff
@@ -0,0 +1,168 @@
+//! Network types and implementations.
+
+use crate::sandbox::SandboxData;
+use anyhow::Result;
+use derive_builder::Builder;
+use getset::Getters;
+use std::path::PathBuf;
+
+pub mod cni;
+
+#[derive(Builder)]
+#[builder(pattern = "owned", setter(into))]
+/// Network is the main structure for working with the Container Network Interface.
+/// The implementation `T` can vary and is being defined in the `Pod` trait.
+pub struct Network<T>
+where
+    T: Default,
+{
+    /// Contains all the plain data for the network.
+    data: NetworkData,
+
+    #[builder(default = "T::default()")]
+    /// Trait implementation for the network.
+    implementation: T,
+}
+
+#[derive(Builder, Getters)]
+#[builder(pattern = "owned", setter(into))]
+/// NetworkData holds all the data which will be passed around to the `Network` trait,
+/// too.
+pub struct NetworkData {
+    #[get = "pub"]
+    /// The network name of the default network to be used.
+    default_network_name: String,
+
+    #[get = "pub"]
+    /// The paths to configuration files.
+    config_paths: Vec<PathBuf>,
+
+    #[get = "pub"]
+    /// The paths to binary plugins.
+    plugin_paths: Vec<PathBuf>,
+}
+
+/// Common network behavior trait
+pub trait PodNetwork {
+    /// Initialize the network, which will be usually done once on application startup.
+    fn initialize(&mut self, _: &NetworkData) -> Result<()> {
+        Ok(())
+    }
+
+    /// Start a new network for the provided `SandboxData`.
+    fn start(&mut self, _: &NetworkData, _: &SandboxData) -> Result<()> {
+        Ok(())
+    }
+
+    /// Stop the network of the provided `SandboxData`.
+    fn stop(&mut self, _: &NetworkData, _: &SandboxData) -> Result<()> {
+        Ok(())
+    }
+
+    /// Ready returns true if the network is ready. This information will be used by the kubelet
+    /// to tell the user the runtime network status.
+    fn ready(&mut self, _: &NetworkData) -> bool {
```

Comment:
This should be tied to the plugin level.

Reply:
Done, we now do not have generic `NetworkData` any more and moved the CNI specific flags into the `CNI` type, too. This way we can later on exchange the implementation by a separate CLI flag.